### PR TITLE
py-fipy:  update to version 3.4.5

### PR DESCRIPTION
**Pull request by jjstickel**
* update to version 3.4.5
* remove post-destroot step (pending approval and cleanup)

#### Description
closes https://trac.macports.org/ticket/70303

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 14.5 23F79 arm64
Xcode 15.4 15F31d

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: python/py-fipy/Portfile [hunk 1]
```diff
@@ -34,9 +34,12 @@ if {${name} ne ${subport}} {
...
                         port:py${python.version}-future
 
-    post-destroot {
-        # Fix permissions problems on these files
-        xinstall -m 0644 -W ${worksrcpath}/FiPy.egg-info PKG-INFO SOURCES.txt dependency_links.txt entry_points.txt top_level.txt \
-            ${destroot}${python.pkgd}/FiPy-${version}-py${python.branch}.egg-info
-    }
+    ## this xinstall command is now causing an error -- I did not implement it,
+    ## and so I am not sure what is wrong with it, nor whether it is even still
+    ## needed, JJS 6/28/24
```
Comment (jmroot) [inline python/py-fipy/Portfile:39]: `.egg-info` is no longer used in modern python packaging, so you can just delete this. The replacement is the `.dist-info` directory, so if you want to check that the issue is gone, you can check the permissions on the files installed in there.

Comment (jonCAstickel) [reply python/py-fipy/Portfile:39]: Looks OK to me:
```
$ ll /opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/FiPy-3.4.5.dist-info/
total 24
-rw-r--r-- 1 root wheel  1878 Jun 28 16:00 LICENSE.rst
-rw-r--r-- 1 root wheel  1838 Jun 28 16:00 METADATA
-rw-r--r-- 1 root wheel 26527 Jun 28 16:00 RECORD
-rw-r--r-- 1 root wheel    91 Jun 28 16:00 WHEEL
-rw-r--r-- 1 root wheel   205 Jun 28 16:00 entry_points.txt
-rw-r--r-- 1 root wheel    12 Jun 28 16:00 top_level.txt
```

**Comment by jonCAstickel** [comment]
This is should be good to merge now. I tested the `mesh1D` example from the source repo, and it worked.

**Comment by jjstickel** [comment]
This is should be good to merge now. I tested the mesh1D example from the source repo, and it worked.

File: python/py-fipy/Portfile (hunk 1 above)
Comment (jjstickel) [reply python/py-fipy/Portfile]: Ok, permissions look good to me:
```
$ ll /opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/FiPy-3.4.5.dist-info/
total 24
-rw-r--r-- 1 root wheel  1878 Jun 28 16:00 LICENSE.rst
-rw-r--r-- 1 root wheel  1838 Jun 28 16:00 METADATA
-rw-r--r-- 1 root wheel 26527 Jun 28 16:00 RECORD
-rw-r--r-- 1 root wheel    91 Jun 28 16:00 WHEEL
-rw-r--r-- 1 root wheel   205 Jun 28 16:00 entry_points.txt
-rw-r--r-- 1 root wheel    12 Jun 28 16:00 top_level.txt
```

**Pull request merged**